### PR TITLE
Report preview

### PR DESCRIPTION
**Pull request by TheLillyleaf**
Implement isly

### CODE REVIEW

File: Report/Expense/PreviewData.ts [hunk 1]
```diff
@@ -1,19 +1,16 @@
 import { isoly } from "isoly"
 import { isly } from "isly"
-import { Amount } from "../../Amount"
+import { Purchase } from "./../../Purchase"
 
 export interface PreviewData {
```
Comment (EliasEriksson) [inline Report/Expense/PreviewData.ts:5]: is it enough to call it preview?

File: Report/Expense/PreviewData.ts [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
-	compileData: Record<string, { purpose: string; date: isoly.Date; amount: Amount }[]>
+	compileData: Record<string, Purchase[] | undefined>
```
Comment (EliasEriksson) [inline Report/Expense/PreviewData.ts:6]: `purchases` or `userPurchases`

File: Report/Receipt/PreviewData.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
 import { isoly } from "isoly"
 import { isly } from "isly"
+import { CostCenter } from "../../CostCenter"
 import { Purchase } from "../../Purchase"
 import { Receipt } from "../../Receipt"
 
 export interface PreviewData {
```
Comment (EliasEriksson) [inline Report/Receipt/PreviewData.ts:7]: is it enough to call it preview?

File: Report/Receipt/PreviewData.ts [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 	receiptsData: {
```
Comment (EliasEriksson) [inline Report/Receipt/PreviewData.ts:8]: is `costCenters` a better name?

**Review by EliasEriksson — COMMENTED** [review]
only name change suggestions

File: Report/Expense/PreviewData.ts (hunk 2 above)
Comment (TheLillyleaf) [reply Report/Expense/PreviewData.ts:6]: Changed it to userExpenses

**Review by EliasEriksson — APPROVED** [review]

**Pull request merged**